asg/ecs.tf

```diff
@@ -23,9 +24,51 @@ resource "aws_ecs_service" "webapp_service" {
     }
 }
 
+resource "aws_ecs_service" "webapp_service_fargate" {
+    count = "${var.launch_type == "FARGATE" ? 1 : 0}"
+    name = "${var.name_prefix}_webapp_service"
+    cluster = "${aws_ecs_cluster.webapp_cluster.id}"
+    task_definition = "${aws_ecs_task_definition.webapp_definition_fargate.arn}"
+    launch_type = "FARGATE"
+    desired_count = "${var.count_webapp}"
+    deployment_minimum_healthy_percent = "${var.minimum_healthy_percent_webapp}"
+    iam_role = "${var.ecs_service_role}"
+
+    network_configuration {
+        security_groups = ["${var.sg_webapp_instances_id}"]
+        subnets         = ["${split(",", var.subnet_ids)}"]
+    }
```

Comment:
For this example, we need to add `assign_public_ip = "true"` for Fargate (since we don't create NAT). Otherwise, our tasks won't be able to pull Docker image from ECR or public Docker Hub. It will return:

```
CannotPullContainerError: API error (500): Get https://registry-1.docker.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
```

See samuelkarp answer [here](https://github.com/aws/amazon-ecs-agent/issues/1128#issuecomment-359517950).

> you either need private IP + NAT or a public IP + IGW. In your example, the task that's failing has neither NAT nor a public IP.



Terraform documentation: https://www.terraform.io/docs/providers/aws/r/ecs_service.html#assign_public_ip